pkg/output_formats/output_formats.go

```diff
@@ -0,0 +1,55 @@
+// Package for having structured access to our output formats
+package output_formats
+
+import (
+	"errors"
+	"strings"
+)
+
+type OutputFormat string
+
+const (
+	Default       = OutputFormat("default")
+	Codeclimate   = OutputFormat("codeclimate")
+	GCC           = OutputFormat("gcc")
+	GithubActions = OutputFormat("github-actions")
+)
+
+var ValidOutputFormats = []OutputFormat{
+	Default,
+	Codeclimate,
+	GCC,
+	GithubActions,
+}
+
+func GetArgumentChoiceText() string {
+	var output_strings []string
+	for _, f := range ValidOutputFormats {
+		output_strings = append(output_strings, string(f))
+	}
+	return strings.Join(output_strings, ", ")
+}
+
+func (format OutputFormat) MarshalText() ([]byte, error) {
+	if !format.IsValid() {
+		return nil, errors.New(string(format) + " is not a valid output format")
+	}
+	return []byte(format), nil
+}
+
+func (format *OutputFormat) UnmarshalText(data []byte) error {
+	*format = OutputFormat(string(data))
+	if !format.IsValid() {
+		return errors.New(string(data) + " is not a valid output format")
+	}
+	return nil
+}
+
+func (format OutputFormat) IsValid() bool {
+	for _, supportedFormat := range ValidOutputFormats {
+		if format == supportedFormat {
+			return true
+		}
+	}
+	return false
```

Comment:
this is [slices.Contains](https://pkg.go.dev/slices#Contains)

Reply:
How TF did miss that while googling explicitly for it... 🤦🏻

Reply:
slices package is somehow one year old, Internet is full of example on how to code it (but posted before slices.Contains existed)

Reply:
hm.. I should dig up that stackoverflow I read and fix it.. but anyway, I replaced it